sdk/core/azure-core/src/test/java/com/azure/core/http/rest/PagedIterableTest.java

```diff
@@ -250,4 +256,90 @@ int getNextPageRetrievals() {
             return nextPageRetrievals;
         }
     }
+
+    @Test
+    public void streamFindFirstOnlyRetrievesOnePage() throws InterruptedException {
+        OnlyOnePageRetriever pageRetriever = new OnlyOnePageRetriever();
+        OnlyOnePagedIterable pagedIterable = new OnlyOnePagedIterable(new OnlyOnePagedFlux(() -> pageRetriever));
+
+        // Validation that there is more than one paged in the full return.
+        pagedIterable.stream().count();
+        int fullPageCount = pageRetriever.getGetCount();
+        assertTrue(fullPageCount > 1);
+
+        Integer next = pagedIterable.stream().findFirst().get();
+
+        Thread.sleep(2000);
+
+        /*
+         * Given that each page contains more than one element we are able to only retrieve a single page.
+         */
+        assertEquals(1, pageRetriever.getGetCount() - fullPageCount);
+    }
```

Comment:
should we add a test for using `pagedIterable.stream().parallel()` and iteration on output of parallel? just double confirming our iterable impl is stream safe.

Reply:
Will add tests for these